[TKET-1915] Add PhasePolyBox creation from circuit to Python (#315)
CQCL/tket: pytket/docs/changelog.rst

Minor new features:

> * Phasepolyboxes can be constructed from a given circuit

Reviewer: ```suggestion
* Phase polynomial boxes can be constructed from a given circuit
```
Please give the method name.

Author: The way to do this is just to use the constructor do you mean that?

Reviewer: Right, something like "new constructor for `PhasePolyBox` given a circuit" -- i.e. mentioning the name of the affected class explicitly.

Author: Done in [](https://github.com/cqc-melf)@cqc-melf
[upadte changelog](https://github.com/CQCL/tket/pull/315/commits/40daea806485ac99fb45f5ce794b21c4ff3ca726)
[40daea8](https://github.com/CQCL/tket/pull/315/commits/40daea806485ac99fb45f5ce794b21c4ff3ca726)

> * New compilerpass ``ComposePhasePolyBoxes`` for generating

Reviewer: ```suggestion
* New compilation pass ``ComposePhasePolyBoxes`` for generating
```

  PhasePolyBoxes in a given circuit
* Add JSON serialization methods for ``Predicate``, ``MeasurementSetup`` and ``MeasurementBitMap``.

Fixes:

* Fix qubit order in ``QubitPauliOperator.to_sparse_matrix()``.
* Fix issue with "nan" values appearing after symbolic substitution following
  compilation of some symbolic circuits.

> * PhasePolyBox constructor is not accepting unvalid boxes anymore

Reviewer: ```suggestion
* PhasePolyBox constructor is not accepting invalid boxes anymore
```
